tracetools/src/tracetools.c

```diff
@@ -362,6 +373,76 @@ void TRACEPOINT(
     handle);
 }
 
+void TRACEPOINT(
+  IPB_to_subscription,
+  const void * IPB,
+  const void * subscription)
+{
+  CONDITIONAL_TP(
+    IPB_to_subscription,
+    IPB,
+    subscription);
+}
+
+void TRACEPOINT(
+  buffer_to_typedIPB,
+  const void * buffer,
+  const void * IPB)
+{
+  CONDITIONAL_TP(
+    buffer_to_typedIPB,
+    buffer,
+    IPB);
+}
+
+void TRACEPOINT(
+  construct_ring_buffer,
+  const void * buffer,
+  const int64_t capacity)
+{
+  CONDITIONAL_TP(
+    construct_ring_buffer,
+    buffer,
+    capacity);
+}
+
+void TRACEPOINT(
+  ring_buffer_enqueue,
+  const void * buffer,
+  const int64_t message,
+  const int64_t accumulated_data,
+  const bool overwriting_occurred)
+{
+  CONDITIONAL_TP(
+    ring_buffer_enqueue,
+    buffer,
+    index,
+    accumulated_data,
+    overwriting_occurred);
```

Comment:
We're supposed to use `index` and not `message` here.

```suggestion
  const uint64_t index,
  const uint64_t size,
  const bool overwritten)
{
  CONDITIONAL_TP(
    ring_buffer_enqueue,
    buffer,
    index,
    size,
    overwritten);
```